server/src/Ogmios/App/Protocol/TxSubmission.hs

```diff
@@ -156,26 +186,31 @@ newExecutionUnitsEvaluator = do
     evaluateExecutionUnitsResponse <- atomically newEmptyTMVar
     return
         ( ExecutionUnitsEvaluator
-            { evaluateExecutionUnitsM = \utxo -> \case
-                GenTxByron{} ->
+            { evaluateExecutionUnitsM = \case
+                (_, GenTxByron{}) ->
                     return (incompatibleEra "Byron")
-                GenTxShelley{} ->
+                (_, GenTxShelley{}) ->
                     return (incompatibleEra "Shelley")
-                GenTxAllegra{} ->
+                (_, GenTxAllegra{}) ->
                     return (incompatibleEra "Allegra")
-                GenTxMary{} ->
+                (_, GenTxMary{}) ->
                     return (incompatibleEra "Mary")
-                GenTxAlonzo (ShelleyTx _id tx) -> do
-                    atomically $ putTMVar evaluateExecutionUnitsRequest (utxo, tx)
+                (SomeUTxOInAlonzoEra utxo, GenTxAlonzo (ShelleyTx _id tx)) -> do
+                    atomically $ putTMVar evaluateExecutionUnitsRequest $ SomeEvaluationInAnyEra utxo tx
+                    atomically $ takeTMVar evaluateExecutionUnitsResponse
+                (SomeUTxOInBabbageEra utxo, GenTxBabbage (ShelleyTx _id tx)) -> do
+                    atomically $ putTMVar evaluateExecutionUnitsRequest $ SomeEvaluationInAnyEra utxo tx
                     atomically $ takeTMVar evaluateExecutionUnitsResponse
```

Comment:
TODO: could/should translate Alonzo to Babbage here in case of discrepancy. Especially because transactions and utxos will parse as old eras first, so it is possible to submit a Babbage transaction with an Alonzo UTxO. In which case, we'll consider the most recent era -> Babbage.